### ISSUE TITLE
Release note workflow

### ISSUE THREAD

**Issue by jakirkham** [zarr-developers/zarr-python#1266]
Currently we label PRs without release notes. Though wonder if we should actually have a failing CI check for this. That might improve visibility and avoid accidentally merging with out release notes. Though unsure how much we want to enforce having release notes (maybe there are cases we don't want to bother with them?). Thoughts?

**Comment by jakirkham** [comment]
cc @Saransh-cpp @MSanKeys963 @joshmoore

**Comment by Saransh-cpp** [comment]
> maybe there are cases we don't want to bother with them?

Yes, cases like minor spelling fix maybe? It won't be nice to fail the check on a PR containing a single spelling fix.

**Comment by joshmoore** [comment]
@DimitriPapadopoulos can correct me, but the criterion he seems to use is "does it touch real code?"

**Comment by DimitriPapadopoulos** [comment]
Good question. I do not really have a formal definition.

Perhaps I would typically not include whatever does not impact functionality, either for ends users or developers, or is not related to a specific release. For example, I would leave out:
- spelling fixes, unless they are important to end users or developers (for example "_use `rm -i` to avoid data loss_" instead of "_use `rm -f` to avoid data loss_" in a user guide)
- running black to format the source code
- changes to CI, perhaps such changes are usually enforced immediately and are therefore not tied to a specific release, rather to a specific commit

I would include some changes not related to actual code:
- change of license, because that does impact end users
- reworked documentation
- changes in data used for testing

**Comment by maxrjones** [comment]
I think release notes automation has been going reasonably well with towncrier, so I'm going to close this issue as completed
